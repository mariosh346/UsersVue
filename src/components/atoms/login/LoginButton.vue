<script setup lang="ts">
import { ref } from 'vue';
import { useAuthFunctions } from 'src/composables/useAuthFunctions';
import BaseBtn from '../BaseBtn.vue';

const { signinPopup } = useAuthFunctions();
const error = ref<unknown>(null);

async function signinRedirect() {

  try {

    await signinPopup();

  } catch (e: unknown) {

    error.value = e;
    console.error('Signin error:', e);

  }

}
</script>

<template>
  <BaseBtn
    icon="login"
    @click="signinRedirect()"
  >
    {{ $t('SignIn with Google') }}
  </BaseBtn>
</template>
