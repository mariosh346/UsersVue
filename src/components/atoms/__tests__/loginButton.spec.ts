import { describe, it, expect, beforeEach, vi } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import LoginButton from "../login/LoginButton.vue";

vi.mock('../composables/useAuthFunctions', () => ({
  useAuthFunctions: () => ({
    signinPopup: vi.fn(() => Promise.resolve())
  })
}));

describe('LoginButton', () => {
  let wrapper: ReturnType<typeof shallowMount>;
  const createWrapper = () => shallowMount(LoginButton);

  beforeEach(() => {
    wrapper = createWrapper();
  });
  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
