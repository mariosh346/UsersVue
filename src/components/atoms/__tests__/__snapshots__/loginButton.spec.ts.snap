// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginButton > matches snapshot 1`] = `"<base-btn-stub icon="login" replace="false" outline="false" flat="false" unelevated="false" rounded="false" push="false" square="false" glossy="false" fab="false" fabmini="false" nocaps="false" nowrap="false" dense="false" ripple="false" stack="false" stretch="false" loading="false" disable="false" round="false" darkpercentage="false"></base-btn-stub>"`;
